evidence/pool.go

```diff
@@ -197,12 +205,19 @@ func (evpool *Pool) CheckEvidence(evList types.EvidenceList) error {
 		ok := evpool.fastCheck(ev)
 
 		if !ok {
+			// check that the evidence isn't already committed
+			if evpool.isCommitted(ev) {
+				return &types.ErrInvalidEvidence{Evidence: ev, Reason: errors.New("evidence was already committed")}
+			}
+
 			evInfo, err := evpool.verify(ev)
 			if err != nil {
 				return &types.ErrInvalidEvidence{Evidence: ev, Reason: err}
 			}
 
 			if err := evpool.addPendingEvidence(evInfo); err != nil {
```

Comment:
Yes you are correct. It doesn't mutate the evidence it just saves it if it hasn't seen it before. This makes it quicker later on when we want to create abci evidence and send it to the application because the evidence pool has already gone and worked out who the validators involved were